libs/src/services/ethContracts/index.js

```diff
@@ -227,18 +234,24 @@ class EthContracts {
     if (discProvTimestamp) {
       try {
         const { endpoint: latestEndpoint, timestamp } = JSON.parse(discProvTimestamp)
-        if ((Date.now() - timestamp) < (5 /* min */ * 60 /* seconds */ * 1000 /* millisec */)) {
+        const inWhitelist = !whitelist || whitelist.has(latestEndpoint)
```

Comment:
@SidSethi could you comment this? these are two different conditions

Reply:
I made these changes
!whitelist means no whitelist was provided (so this check should satisfy)
whitelist.has(latestEndpoint) means a whilelist was provided and value is in the whitelist

perhaps inWhitelistOrNoWhitelist is the better var name